### DISCUSSION TITLE
Clarification on the function "compute_band_power_deviation_and_tilt" in GnPy 2.11

### DISCUSSION THREAD

**Discussion by mukherjeearup** [Telecominfraproject/oopt-gnpy#520] [Q&A]
Hi,
I see the following method used to find SRS tilt target and relative power deviation at each band:

def compute_band_power_deviation_and_tilt(srs_power_deviation, design_bands: dict, ratio: float = 0.8):
    """Compute the power difference between bands (at center frequency) and the power tilt within each
    band.

    Args:
        srs_power_deviation: The list of dictionnary containing the power at band centers and the tilt within each band.
        ratio: the ratio applied to compute the band tilt
    Returns:
        A tupple of dict containing the relative power deviation with respect to max value, per band in dB and the tilt
        target to apply for each band.
    """

I have couple of questions:

1. In the above method, Arg "srs_power_deviation" contains each band centre freq power and tilt within each band, but I did not understand, how is the other arg "ratio" derived with the default value of 0.8?
2. Because of SRS, the centre freq of each band will have different power. How is that compensated in the gain target of the next amplifier?

Regards,
Arup

**Comment by EstherLerouzic** [comment]
Dear @mukherjeearup,

Thank you for your interest in GNPy! 

The srs_power_deviation argument indeed contains the power levels at the center frequencies of each band, as well as the tilt within each band. The ratio parameter, which defaults to 0.8, is a tuning factor that helps adjust the compensation applied to the amplifier gain target based on the estimated power deviation. This ratio was chosen somewhat arbitrarily, but it can be modified by the user to achieve the desired level of compensation for SRS effects.

In terms of how the different power levels at the band center frequencies are compensated in the gain target of the amplifier, the strategy primarily depends on the specific goals of the network design. For instance, the desired outcome could be to achieve flat power distribution across bands, optimize for flat GSNR, or prioritize performance in the C band over the L band. The power deviation estimated from the previous and next spans surrounding the amplifiers is crucial in this process.

Currently, our implementation uses a straightforward model where the SRS tilt is compensated based on the power deviation from the previous span only. The computed srs_power_deviation is then utilized to set the amplifier gain for each band. The ratio of 0.8 serves as a multiplier to scale the compensation, allowing for flexibility in tuning the amplifier settings according to user preferences.

Looking ahead, we plan to enhance this functionality by incorporating advanced optimization techniques based on the findings from the paper: V. Curri, B. Correia, and A. D'Amico, "Challenges in Modeling and Controlling WDM Transmission in Multiband Optical Networks," presented at the 2023 International Conference on Photonics in Switching and Computing (PSC). This will provide users with more sophisticated options for managing SRS effects and amplifier settings.

You can find the paper here: https://ieeexplore.ieee.org/abstract/document/10297164.

If you have further questions or need more clarification, feel free to ask!
